### ISSUE TITLE
[1.21] Crash on fabric server

### ISSUE THREAD

**Issue by Hiusi** [LeoMinecraftModding/Eternal-Starlight#2]
> Fabric 1.21 0.16.0
> Eternal version 0.1.2+1.21.1+fabric

Server cannot start and produce this error
```
[00:28:54] [main/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.util.ServiceConfigurationError: cn.leolezury.eternalstarlight.common.platform.ESPlatform: cn.leolezury.eternalstarlight.fabric.platform.FabricPlatform Unable to get public no-arg constructor
        at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:63) ~[fabric-loader-0.16.0.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:482) ~[fabric-loader-0.16.0.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.0.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.16.0.jar:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.16.0.jar:?]
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [server.jar:1.0.1]
Caused by: java.util.ServiceConfigurationError: cn.leolezury.eternalstarlight.common.platform.ESPlatform: cn.leolezury.eternalstarlight.fabric.platform.FabricPlatform Unable to get public no-arg constructor
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:586) ~[?:?]
        at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:679) ~[?:?]
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1240) ~[?:?]
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273) ~[?:?]
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309) ~[?:?]
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393) ~[?:?]
        at cn.leolezury.eternalstarlight.common.platform.ESPlatform.lambda$static$0(ESPlatform.java:56) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at net.minecraft.class_156.method_656(class_156.java:533) ~[server-intermediary.jar:?]
        at cn.leolezury.eternalstarlight.common.platform.ESPlatform.<clinit>(ESPlatform.java:53) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at cn.leolezury.eternalstarlight.common.platform.registry.RegistrationProvider.get(RegistrationProvider.java:11) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at cn.leolezury.eternalstarlight.common.registry.ESAttributes.<clinit>(ESAttributes.java:11) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at net.minecraft.class_1309.handler$bkj000$eternal_starlight$createLivingAttributes(class_1309.java:18343) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_26827(class_1309.java:319) ~[server-intermediary.jar:?]
        at net.minecraft.class_1308.method_26828(class_1308.java:172) ~[server-intermediary.jar:?]
        at net.minecraft.class_7298.method_42655(class_7298.java:177) ~[server-intermediary.jar:?]
        at net.minecraft.class_5135.<clinit>(class_5135.java:91) ~[server-intermediary.jar:?]
        at net.minecraft.class_2966.method_17598(class_2966.java:134) ~[server-intermediary.jar:?]
        at net.minecraft.server.Main.main(Main.java:109) ~[server-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) ~[fabric-loader-0.16.0.jar:?]
        ... 4 more
Caused by: java.lang.NoClassDefFoundError: net/minecraft/class_1087
        at java.base/java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
        at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3549) ~[?:?]
        at java.base/java.lang.Class.getConstructor0(Class.java:3754) ~[?:?]
        at java.base/java.lang.Class.getConstructor(Class.java:2442) ~[?:?]
        at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:666) ~[?:?]
        at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:663) ~[?:?]
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:571) ~[?:?]
        at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:674) ~[?:?]
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1240) ~[?:?]
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273) ~[?:?]
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309) ~[?:?]
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393) ~[?:?]
        at cn.leolezury.eternalstarlight.common.platform.ESPlatform.lambda$static$0(ESPlatform.java:56) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at net.minecraft.class_156.method_656(class_156.java:533) ~[server-intermediary.jar:?]
        at cn.leolezury.eternalstarlight.common.platform.ESPlatform.<clinit>(ESPlatform.java:53) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at cn.leolezury.eternalstarlight.common.platform.registry.RegistrationProvider.get(RegistrationProvider.java:11) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at cn.leolezury.eternalstarlight.common.registry.ESAttributes.<clinit>(ESAttributes.java:11) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at net.minecraft.class_1309.handler$bkj000$eternal_starlight$createLivingAttributes(class_1309.java:18343) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_26827(class_1309.java:319) ~[server-intermediary.jar:?]
        at net.minecraft.class_1308.method_26828(class_1308.java:172) ~[server-intermediary.jar:?]
        at net.minecraft.class_7298.method_42655(class_7298.java:177) ~[server-intermediary.jar:?]
        at net.minecraft.class_5135.<clinit>(class_5135.java:91) ~[server-intermediary.jar:?]
        at net.minecraft.class_2966.method_17598(class_2966.java:134) ~[server-intermediary.jar:?]
        at net.minecraft.server.Main.main(Main.java:109) ~[server-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) ~[fabric-loader-0.16.0.jar:?]
        ... 4 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_1087
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226) ~[fabric-loader-0.16.0.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.0.jar:?]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        at java.base/java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
        at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3549) ~[?:?]
        at java.base/java.lang.Class.getConstructor0(Class.java:3754) ~[?:?]
        at java.base/java.lang.Class.getConstructor(Class.java:2442) ~[?:?]
        at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:666) ~[?:?]
        at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:663) ~[?:?]
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:571) ~[?:?]
        at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:674) ~[?:?]
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1240) ~[?:?]
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273) ~[?:?]
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309) ~[?:?]
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393) ~[?:?]
        at cn.leolezury.eternalstarlight.common.platform.ESPlatform.lambda$static$0(ESPlatform.java:56) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at net.minecraft.class_156.method_656(class_156.java:533) ~[server-intermediary.jar:?]
        at cn.leolezury.eternalstarlight.common.platform.ESPlatform.<clinit>(ESPlatform.java:53) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at cn.leolezury.eternalstarlight.common.platform.registry.RegistrationProvider.get(RegistrationProvider.java:11) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at cn.leolezury.eternalstarlight.common.registry.ESAttributes.<clinit>(ESAttributes.java:11) ~[eternalstarlight-0.1.2+1.21.1+fabric.jar:?]
        at net.minecraft.class_1309.handler$bkj000$eternal_starlight$createLivingAttributes(class_1309.java:18343) ~[server-intermediary.jar:?]
        at net.minecraft.class_1309.method_26827(class_1309.java:319) ~[server-intermediary.jar:?]
        at net.minecraft.class_1308.method_26828(class_1308.java:172) ~[server-intermediary.jar:?]
        at net.minecraft.class_7298.method_42655(class_7298.java:177) ~[server-intermediary.jar:?]
        at net.minecraft.class_5135.<clinit>(class_5135.java:91) ~[server-intermediary.jar:?]
        at net.minecraft.class_2966.method_17598(class_2966.java:134) ~[server-intermediary.jar:?]
        at net.minecraft.server.Main.main(Main.java:109) ~[server-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) ~[fabric-loader-0.16.0.jar:?]
        ... 4 more
container@dolphin~ Server marked as offline...
```

**Comment by LeoLezury** [comment]
Thanks. It's fixed now, I'll publish a new version soon.

**Comment by LeoLezury** [comment]
Fixed in 0.1.3👍Now available on **modrinth**.
